src/sidebar/components/MarkdownEditor.js

```diff
@@ -19,39 +19,63 @@ import { isMacOS } from '../../shared/user-agent';
 
 import MarkdownView from './MarkdownView';
 
+/**
+ * @template T
+ * @typedef {import('preact').RefObject<T>} Ref
+ */
+
 /**
  * @typedef {import("@hypothesis/frontend-shared/lib/components/Link").LinkProps} LinkProps
  * @typedef {import('preact').JSX.HTMLAttributes<HTMLTextAreaElement>} TextAreaAttributes
- * @typedef {import('preact').Ref<HTMLTextAreaElement>} TextAreaRef
+ * @typedef {import('../markdown-commands').EditorState} EditorState
+ */
+
+/**
+ * Toolbar commands that modify the editor state. This excludes the Help link
+ * and Preview buttons.
  *
- * @typedef {'bold'|'italic'|'quote'|'link'|'image'|'math'|'numlist'|'list'|'preview'|'help'} ButtonID
+ * @typedef {'bold'|
+ *   'image'|
+ *   'italic'|
+ *   'link'|
+ *   'list' |
+ *   'math'|
+ *   'numlist'|
+ *   'quote'
+ * } Command
```

Comment:
This is a little awkwardly vertical :) but NBD.

Reply:
Unfortunately TS fails to parse the `@typedef` unless the first element of the union is written on the first line. Keeping the rest of it vertical makes it easy to slot in new items and re-sort though.